portals/publisher/src/main/webapp/source/src/app/components/GlobalPolicies/Listing/Listing.tsx

```diff
@@ -0,0 +1,1115 @@
+                return {
+                    ...policy,
+                    appliedGatewayLabels: [...policy.appliedGatewayLabels, ...newLabels],
+                };
+            }
+            return policy;
+        });
+        setPolicies(updatedPolicies);
+        setSelectedGateways(cleanSeletectedGateways(policyId));
+    };
+
+    /**
+     * Function to remove a label from a specific policy in UI.
+     * @param {string} policyId - Policy Identifier.
+     * @param {string} labelToRemove - Removing enviroment/gateway from applied Gateway Labels.
+     */
+    const removeLabelFromPolicy = (policyId: string, labelToRemove: string) => {
+        const updatedPolicies = policies.map((policy) => {
+            if (policy.id === policyId) {
+                return {
+                    ...policy,
+                    appliedGatewayLabels: policy.appliedGatewayLabels.filter((label) => label !== labelToRemove),
+                };
+            }
+            return policy;
+        });
+        setPolicies(updatedPolicies);
+        setSelectedGateways(cleanSeletectedGateways(policyId));
+    };
+
+    /**
+     * Function to undeploy a policy mapping to another enviroment/gateway.
+     * @param {string} gatewayPolicyMappingId - Policy Identifier.
+     * @param {string} environement - Deploying enviroment/gateway.
+     * @param {boolean} deploying - Deploying or undeploying.
+     */
+    const undeploy = (gatewayPolicyMappingId: string, environement: string) => {
+        setIsUndeployDialogOpen(false);
+        setLoading(true);
+        const deploymentArray = getDeploymentArray(gatewayPolicyMappingId);
+        const updatedDeploymentArray = toggleGatewayDeployment(deploymentArray, environement);
+        /**
+         * call the backend API and handle the response
+         */
+        const promise = API.engageGlobalPolicy(gatewayPolicyMappingId, updatedDeploymentArray);
+        promise
+            .then((response) => {
+                if (response.status === 200 || response.status === 201) {
+                    setLoading(false);                  
+                    APIMAlert.success(intl.formatMessage({
+                        id: 'Policy.Undeploy.Successful',
+                        defaultMessage: 'Policy undeployed successfully',
+                    }));    
+                    /**
+                     * If successful, remove from the state rather than getting from backend.
+                     */
+                    removeLabelFromPolicy(gatewayPolicyMappingId, environement);                                
+                }
+                else {
+                    APIMAlert.error(intl.formatMessage({
+                        id: 'Error.Deploy.Policy',
+                        defaultMessage: 'Error occurred while deploying the policy',
+                    }));
+                }                
+            })
+            .catch((/* error */) => {
+                // console.error(error);
```

Comment:
Remove commented code

Reply:
Addressed the required changes by [fddf57f](https://github.com/wso2/apim-apps/pull/515/commits/fddf57fa7308a53094c808e9bf2d2b40ed71b644) and [0ad490c](https://github.com/wso2/apim-apps/pull/515/commits/0ad490c497b08f999babbe521feaf83d029a6698)